sql/stmtdiagnostics: `TestDiagnosticsRequest` failures under remote execution

EngFlow link: https://tanzanite.cluster.engflow.com/invocation/fc95e211-60da-4fd1-b927-44b2fa393af7
Reproduction command: `bazel test //pkg/sql/stmtdiagnostics:stmtdiagnostics_test --config engflow --config crosslinux --jobs 400 --config cibase --test_filter=TestDiagnosticsRequest --test_arg -show-logs`

https://cockroachlabs.atlassian.net/wiki/spaces/devinf/pages/3141107902/How+to+reproduce+a+test+failure+on+EngFlow
Epic CRDB-14596


Jira issue: CRDB-30812

I took a stab at this with running `bazel test //pkg/sql/stmtdiagnostics:stmtdiagnostics_test --config engflow --config crosslinux --jobs 400 --config cibase --test_filter=TestDiagnosticsRequest --test_arg -show-logs --test_arg -test.v --test_arg --test.count=10`. The first such run succeeded, but on the second I got some [failures](https://tanzanite.cluster.engflow.com/invocation/62108c67-f234-4bdd-931f-a1b9bd8a3f05):
```
    statement_diagnostics_test.go:293: condition failed to evaluate within 45s: from statement_diagnostics_test.go:327: expected request to be completed
        --- FAIL: TestDiagnosticsRequest/ongoing_request_canceled/conditional
...
    statement_diagnostics_test.go:293: condition failed to evaluate within 45s: from statement_diagnostics_test.go:327: expected request to be completed
        --- FAIL: TestDiagnosticsRequest/ongoing_request_canceled/conditional#01
```
This type of failure is definitely SQL Queries although we haven't seen this happen in non-remote execution mode ever (?).

--- a/pkg/ccl/multitenantccl/tenantcostclient/BUILD.bazel
+++ b/pkg/ccl/multitenantccl/tenantcostclient/BUILD.bazel
@@ -51,7 +51,6 @@ go_test(
     ],
     data = glob(["testdata/**"]),
     embed = [":tenantcostclient"],
-    tags = ["no-remote-exec"],
     deps = [
         "//pkg/base",
         "//pkg/ccl",

--- a/pkg/sql/stmtdiagnostics/BUILD.bazel
+++ b/pkg/sql/stmtdiagnostics/BUILD.bazel
@@ -40,7 +40,6 @@ go_test(
     ],
     embed = [":stmtdiagnostics"],
     exec_properties = {"test.Pool": "large"},
-    tags = ["no-remote-exec"],
     deps = [
         "//pkg/base",
         "//pkg/kv",

